tests/lib/mem_alloc/testcase.yaml

```diff
@@ -6,5 +6,6 @@ tests:
   libraries.libc.newlib:
     extra_args: CONF_FILE=prj_newlib.conf
     arch_exclude: posix
+    platform_exclude: colibri_imx7d_m4 warp7_m4
```

Comment:
What caused these boards to start failing? And why do platforms with smaller flashes build successfully? Unlike the counter case in #15462, aren't we always building with newlib here?